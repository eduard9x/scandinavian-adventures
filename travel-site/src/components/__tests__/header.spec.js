import React from 'react';

import Header from '../header';
import {describe, expect, it} from '@jest/globals';
import {render, screen} from '@testing-library/react';

describe('Header', () => {
  it('should be defined', () => {
    const pageTitle = 'Default Starter';
    render(
        <Header siteTitle={pageTitle}/>,
    );
    expect(screen.queryByText(pageTitle)).toBeDefined();
  });

  it('links to homepage', () => {
    const pageTitle = 'Default Starter';
    const basePath = 'http://localhost';
    render(
        <Header siteTitle={pageTitle}/>,
    );
    expect(screen.queryByText(pageTitle)).toHaveProperty('href', `${basePath}/`);
  });
});
