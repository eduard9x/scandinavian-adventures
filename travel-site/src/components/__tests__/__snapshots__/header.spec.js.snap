// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="bg-white fixed w-full z-50"
>
  <div
    className="my-0 mx-auto py-6 px-5 justify-between flex items-center h-full"
    style={
      Object {
        "maxWidth": 1280,
      }
    }
  >
    <h3
      className="m-0 inline-flex"
    >
      <a
        className="text-black no-underline focus:outline-black"
        href="/"
      >
        Default Starter
      </a>
    </h3>
    <button
      className="p-1 focus:outline-black "
      onClick={[Function]}
    >
      <div
        className="open"
        id="nav-icon4"
      >
        <span />
        <span />
        <span />
      </div>
    </button>
  </div>
</header>
`;
